crates/bevy_audio/src/audio_output.rs

```diff
@@ -1,16 +1,22 @@
-use crate::AudioSource;
+use crate::Decodable;
 use bevy_asset::{Assets, Handle};
 use bevy_ecs::Res;
-use rodio::{Decoder, Device, Sink};
-use std::{collections::VecDeque, io::Cursor, sync::RwLock};
+use rodio::{Device, Sink};
+use std::{collections::VecDeque, sync::RwLock};
 
 /// Used to play audio on the current "audio device"
-pub struct AudioOutput {
+pub struct AudioOutput<P>
```

Comment:
Can we make this `AudioOutput<P=AudioSource>` to make this a non-breaking change / retain the ease-of-use we had before?